--- a/gcc/Makefile.in
+++ b/gcc/Makefile.in
@@ -2998,7 +2998,7 @@ stmp-fixinc: gsyslimits.h macro_list fix
 	      echo "  ${SYSTEM_HEADER_DIR}" >&2 ; \
 	      tooldir_sysinc=`echo "${gcc_tooldir}/sys-include" | sed -e :a -e "s,[^/]*/\.\.\/,," -e ta`; \
 	      if test "x${SYSTEM_HEADER_DIR}" = "x$${tooldir_sysinc}"; \
-	      then sleep 1; else exit 1; fi; \
+	      then sleep 1; else true; fi; \
 	    fi; \
 	    $(mkinstalldirs) $${fix_dir}; \
 	    chmod a+rx $${fix_dir} || true; \
