From 821706e44170ce2c42b3a5ed3fdad85d21f8ebca Mon Sep 17 00:00:00 2001
From: Weijie Gao <hackpascal@gmail.com>
Date: Tue, 28 Jan 2020 20:40:34 +0800
Subject: Honor mingw's native prefix in sysroot

---
 ld/ldmain.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/ld/ldmain.c b/ld/ldmain.c
index 77cdbd0d..bdd656a7 100644
--- a/ld/ldmain.c
+++ b/ld/ldmain.c
@@ -568,6 +568,31 @@ get_relative_sysroot (const char *from ATTRIBUTE_UNUSED)
   char *path;
   struct stat s;
 
+#ifdef MINGW_NATIVE_PREFIX
+  char *path_mingw;
+  size_t len_sysroot, len_prefix;
+
+  len_sysroot = strlen (TARGET_SYSTEM_ROOT);
+  len_prefix = strlen (MINGW_NATIVE_PREFIX);
+
+  path_mingw = (char *) xmalloc (len_sysroot + len_prefix + 2);
+
+  strcpy (path_mingw, TARGET_SYSTEM_ROOT);
+  if (path_mingw[len_sysroot - 1] == '/')
+    path_mingw[len_sysroot - 1] = 0;
+
+  strcat(path_mingw, MINGW_NATIVE_PREFIX);
+
+  path = make_relative_prefix (program_name, from, path_mingw);
+  free (path_mingw);
+  if (path)
+    {
+      if (stat (path, &s) == 0 && S_ISDIR (s.st_mode))
+	return path;
+      free (path);
+    }
+#endif
+
   path = make_relative_prefix (program_name, from, TARGET_SYSTEM_ROOT);
   if (path)
     {
-- 
2.17.1

